### PR TITLE
tests(hyper): Add tests to increase coverage

### PR DESCRIPTION
**Pull request by pyfisch**

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: d2f0cea53f0dac3cd4776ee0f79e8f767215355c
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by seanmonstar** [comment]
I'm skeptical that coveralls and/or kcov is correctly reporting coverage, but certainly, more tests are always good!

File: src/header/common/connection.rs [hunk 1]
```diff
@@ -50,38 +25,49 @@ pub enum ConnectionOption {
...
+            s => Ok(ConnectionHeader(UniCase(s.to_string())))
+        }
     }
 }
 
 impl Display for ConnectionOption {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.write_str(match *self {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", match *self {
```
Comment (seanmonstar) [inline src/header/common/connection.rs]: why the switch from `f.write_str`? No actual formatting is needed...

Comment (pyfisch) [inline src/header/common/connection.rs]: My mistake.

**Comment by pyfisch** [comment]
@seanmonstar Yes I also doubt that kcov is working correctly. There are some files even missing in the report. But it reports what cases are not covered by the tests so I fix the reported cases.

**Pull request merged**